lib/fs.js

```diff
@@ -2379,8 +2379,13 @@ WriteStream.prototype.open = function() {
 
 
 WriteStream.prototype._write = function(data, encoding, cb) {
-  if (!(data instanceof Buffer))
-    return this.emit('error', new Error('Invalid data'));
+  if (!(data instanceof Buffer)) {
+    const err = new errors.TypeError('ERR_INVALID_ARG_TYPE',
+                                     'data',
+                                     'Buffer',
+                                     data);
+    return this.emit('error', err);
```

Comment:
while we are at it, I would see if we can convert this to `this.detroy(err)` instead.

Reply:
Hi @mcollina, this is my first time looking at the source code for Node so I'm not sure what the implications are of changing that line 🙃 

1. Do you want me to change `return this.emit('error', err);` to `return this.detroy(err);`?
2. How would I ensure that change is working correctly (besides existing unit tests)?

Thanks! 😄

Reply:
1. yes.

2. run our unit tests. You can check if `_destroy` gets called, but that's a plus.